config/shared_config.go

```diff
@@ -337,26 +793,34 @@ func (c *SharedConfig) setFromIniFiles(profiles map[string]struct{}, profile str
 	return nil
 }
 
-// setFromFile loads the configuration from the file using
-// the profile provided. A SharedConfig pointer type value is used so that
+// setFromIniSection loads the configuration from the profile section defined in
+// the provided ini file. A SharedConfig pointer type value is used so that
 // multiple config file loadings can be chained.
 //
 // Only loads complete logically grouped values, and will not set fields in cfg
 // for incomplete grouped values in the config. Such as credentials. For example
 // if a config file only includes aws_access_key_id but no aws_secret_access_key
 // the aws_access_key_id will be ignored.
-func (c *SharedConfig) setFromIniFile(profile string, file sharedConfigFile) error {
-	section, ok := file.IniData.GetSection(profile)
-	if !ok {
-		// Fallback to to alternate profile name: profile <name>
-		section, ok = file.IniData.GetSection(fmt.Sprintf("profile %s", profile))
-		if !ok {
-			return SharedConfigProfileNotExistError{
-				Filename: file.Filename,
-				Profile:  profile,
-				Err:      nil,
-			}
+func (c *SharedConfig) setFromIniSection(profile string, section ini.Section) error {
+	if len(section.Name) == 0 {
+		sources := make([]string, 0)
+		for _, v := range section.SourceFile {
+			sources = append(sources, v)
+		}
+
+		return SharedConfigProfileNotExistError{
+			Filename: sources,
+			Profile:  profile,
+			Err:      nil,
+		}
+	}
```

Comment:
probably want a different error since this case isn't expected to happen due to the ini section processing above.